src/common/hook/useRevokeQueue/utils/revoke.tsx

```diff
@@ -0,0 +1,31 @@
+import { RevokingJob } from "../../../../types";
+import { utils } from "@govtechsg/open-attestation";
+
+export const getRevokingJobs = async (documents: any[], nonce: number): Promise<RevokingJob[]> => {
+  return Promise.all(
+    documents.map(async (document) => {
+      let contractAddress = "";
+      try {
+        contractAddress = utils.getIssuerAddress(document)[0];
+      } finally {
+        if (!contractAddress) {
+          if (utils.isWrappedV2Document(document)) {
+            const unwrappedDocument = utils.getData(document);
+            const issuer = unwrappedDocument.issuers[0];
+            contractAddress = issuer.revocation?.location || "";
+          } else if (utils.isWrappedV3Document(document)) {
+            contractAddress = document.openAttestationMetadata.proof.revocation?.location || "";
+          }
+        }
+      }
```

Comment:
i think should not even reach this function, for as long as `revocation` with `NONE` value is detected. then here will be:

```suggestion
      let contractAddress = "";
      if (utils.isWrappedV2Document(document)) {
        contractAddress = utils.getIssuerAddress(document)[0];
      } else if (utils.isWrappedV3Document(document)) {
        contractAddress = utils.getIssuerAddress(document);
      }
```

Reply:
erm yeah sure will update it

Reply:
i cant update the getIssuerAddress as if its did, the issuerAddress is not in revocation.location. its the revocation store address, the key doesn't match. so i have to look for it in this function.